chain/chain/src/signature_verification.rs

```diff
@@ -62,3 +64,28 @@ pub fn verify_chunk_header_signature_with_epoch_manager(
         block_info,
     )
 }
+
+/// This function requires that the previous block of `header` has been processed.
+/// If not, it returns EpochError::MissingBlock.
+pub fn verify_header_signature_with_epoch_manager(
+    epoch_manager: &dyn EpochManagerAdapter,
+    header: &BlockHeader,
+) -> Result<bool, Error> {
+    let block_producer =
+        epoch_manager.get_block_producer_info(header.epoch_id(), header.height())?;
+    match epoch_manager.get_block_info(header.prev_hash()) {
+        Ok(block_info) => Ok(verify_header_signature(header, block_producer, block_info)),
+        Err(_) => return Err(EpochError::MissingBlock(*header.prev_hash()).into()),
+    }
+}
+
+fn verify_header_signature(
+    header: &BlockHeader,
+    block_producer: ValidatorStake,
+    block_info: Arc<BlockInfo>,
+) -> bool {
+    if block_info.slashed().contains_key(block_producer.account_id()) {
```

Comment:
I would rather just remove this function and push the functionality inline to `verify_header_signature_with_epoch_manager`.

It's not trivial to imagine this function should be checking slashed block producer from block info of previous block.

Reply:
+1, this slashing check is annoying and useless at this point, would be great to remove it

Reply:
In this case, does it still matter whether the previous block has been processed? More exactly, is [this](https://github.com/near/nearcore/pull/12467/commits/be70b281a271cc3bffa1c68bca6e186719018bf3#diff-86ae05b243b7d9078ac67f1464ca9a35874e84c69c3932216ec05b6040c09e1eL76-L88) change correct?